components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaFetchRecords.java

```diff
@@ -359,6 +369,23 @@ protected void startPolling() {
         }
     }
 
+    private void updateTaskState() {
+        switch (state) {
+            case PAUSE_REQUESTED:
+                LOG.info("Pausing the consumer as a response to a pause request");
+                consumer.pause(consumer.assignment());
+                state = State.PAUSED;
+                break;
+            case RESUME_REQUESTED:
+                LOG.info("Resuming the consumer as a response to a resume request");
+                consumer.resume(consumer.assignment());
+                state = State.RUNNING;
+                break;
+            default:
+                break;
+        }
```

Comment:
I don't know if it is intended but it seems that the states `RUNNING` and `PAUSED` are set but never used. Maybe you can set the state to `null` once done?

Reply:
This is intended, but I think I can do better. I want to be explicit about the state it is.

Reply:
Actually, I think it's better to keep it as is. Otherwise we'll just unnecessarily implement a  state machine without necessity (we can probably get rid of 1 of the states, though).

Reply:
I'll add a comment though. This, indeed, can be confusing.